spec/helpers.lua

```diff
@@ -1254,6 +1255,108 @@ local function proxy_client_grpcs(host, port)
 end
 
 
+---
+-- Reconfiguration completion detection helpers
+--
+
+local MAX_RETRY_TIME = 10
+
+--- Set up admin client and proxy client to so that interactions with the proxy client
+-- wait for preceding admin API client changes to have completed.
+
+-- @function make_synchronized_clients
+-- @param options table with options
+-- @return admin_client, proxy_client
+
+local function make_synchronized_clients(clients)
```

Comment:
I like the design of the API. It's simple yet we can customize the clients. But In some cases, we want more than 1 proxy client(for example to test more than 1 proxying port or to test some of the status observing ports (which is not necessarily a proxy client). Could we generalize it further, like this?

```lua
local admin_client = make_synchronized_clients()
local bound_client1 = admin_client :version_bind(client1)
local bound_client2 = admin_client :version_bind(client2)
```

Reply:
We need the one entry point that synchronizes one proxy and one admin client, but we can add a method to the clients so that you can synchronize other clients afterwards:
```lua
proxy_client, admin_client = helpers.make_synchronized_clients()
admin_client:synchronize_sibling(other_admin_client)
```
Does that make sense?  I'm not entirely happy about `synchronize_sibling`, but `bind` is too generic for my taste.

Reply:
@hanshuebner I was trying to bind a new proxy client (who waits for the config update). I agree that `bind` is too generic. I just cannot come up with a better name. The interface you propose looks OK to me (bind new proxy clients with the admin client).

Reply:
And you reminded me. In some cases admin client inspects the status of Kong (rbac, plugin endpoint, etc), which also needs to be synced with the config change.

Reply:
If you need additional explicit synchronization points in a test, you can always make a dummy proxy request to force it.  I think this should be good enough for now.